test/E2ETest/Infrastructure/EqualityYamlNodeVisitor.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System;
+using Tye.Serialization;
+using YamlDotNet.RepresentationModel;
+
+namespace E2ETest
+{
+    public class EqualityYamlNodeVisitor
+    {
+        public EqualityYamlNodeVisitor()
+        {
+        }
+
+        public void Visit(YamlNode node, YamlNode otherNode)
+        {
+            if (node.NodeType == YamlNodeType.Mapping)
+            {
+                VisitMapping((YamlMappingNode)node, (YamlMappingNode)otherNode);
+            }
+            else if (node.NodeType == YamlNodeType.Scalar)
+            {
+                VisitScalar((YamlScalarNode)node, (YamlScalarNode)otherNode);
+            }
+            else if (node.NodeType == YamlNodeType.Sequence)
+            {
+                VisitSequence((YamlSequenceNode)node, (YamlSequenceNode)otherNode);
+            }
+        }
+
+        public void VisitSequence(YamlSequenceNode node, YamlSequenceNode otherNode)
+        {
+            if (otherNode == null)
+            {
+                throw new TyeYamlException($"Expected sequence to be not null, difference starting at Expected: ({node.Start.Line}, {node.Start.Column})");
+            }
+
+            if (node.Tag?.Equals(otherNode.Tag) == false)
+            {
+                throw new TyeYamlException($"Expected tags to be equal, difference starting at Expected: ({node.Start.Line}, {node.Start.Column}) Actual: ({otherNode.Start.Line}, {otherNode.Start.Column})");
+            }
+
+            if (node.Children.Count != otherNode.Children.Count)
+            {
+                throw new TyeYamlException($"Number of children differ for sequence, Expected: ({node.Start.Line}, {node.Start.Column}) Actual: ({otherNode.Start.Line}, {otherNode.Start.Column})");
+            }
```

Comment:
It might be better to put this at the end, and then you could include things like the text of the *extra* children afterwards.

Reply:
Yeah, I had it here because I didn't want to throw if otherNode had less children. I can break up the check into two places though.

Reply:
Actually, nvrm I see what you are getting at.